### PR TITLE
TST: Fix LiveTable test by using lossless subs.

### PR DESCRIPTION
**Pull request by danielballan**
The LiveTable test was failing because, I think, the lossy subscription was doing its job, dropping an event or two, and thus missing a row.

Also, this makes a non-breaking API change that was previously discussed: `RunEngine.subscribe_lossless` and `RunEngine.unsubscribe_lossless` are now public methods. Aliases for the private names are maintained in case someone (HXN?) is using them.

(These were originally (back in May) made private so that users didn't choke their processors by reaching for lossless subscriptions to liberally. That's not really the way private methods are supposed to be used, and in any case that imagined scenario has not been a problem.)

### CODE REVIEW

**Comment by danielballan** [comment]
Ooof. @tacaswell, Travis fails on

```python
ValueError: Allowed signals are <enum 'DocumentNames'>
```

*Really* regretting our use of a enum instead of strings. Time to walk that back (for next cycle)?

**Comment by danielballan** [comment]
Update: We can postpone paying down the technical debt of DocumentNames enum for another day -- the Dispatcher handles it.

For reasons that no longer apply, the *lossless* subscriptions were handled directly to a CallbackRegistry, not laundered through a Dispatcher (which adds nice validation and convenience methods). Now in this PR we just use two Dispatchers.

Test coverage on this stuff is solid -- I'm not worried about this going in this cycle.

File: bluesky/run_engine.py [hunk 1]
```diff
@@ -251,23 +251,31 @@ def __init__(self, md=None, *, md_validator=None):
...
+        # to provide special docstrings.
+        self._lossless_dispatcher = Dispatcher()
 
         self.verbose = False
 
         loop.call_soon(self._check_for_trouble)
         loop.call_soon(self._check_for_signals)
 
+        # aliases for back-compatibility
+        self._unsubscribe_lossless = self.unsubscribe_lossless
```
Comment (tacaswell) [inline bluesky/run_engine.py]: These should be done at class level as `_unsubscribe_lossless = unsubscribe_lossless`

**Comment by tacaswell** [comment]
What were the reasons that no longer apply?

File: bluesky/run_engine.py (hunk 1 above)
Comment (danielballan) [inline bluesky/run_engine.py]: Good call. Done.

**Comment by danielballan** [comment]
@tacaswell The Dispatcher used to be less configurable than it is today.

**Comment by klauer** [comment]
Looks good

**Pull request merged**